lib/codit/population/networks/city.py

```diff
@@ -6,50 +6,110 @@
 from codit.population.networks import household_workplace
 from codit.population.networks.city_config.city_cfg import MINIMUM_WORKING_AGE, MAXIMUM_WORKING_AGE, MAXIMUM_CLASS_AGE, MINIMUM_CLASS_AGE, AVERAGE_HOUSEHOLD_SIZE
 from codit.population.networks.city_config.typical_households import build_characteristic_households
+# Import home module
+from codit.population.networks import home
+from codit.population.networks.home import Home, COORDINATES_CSV, TYPES_CONSTRAINTS_CSV
 
 EPHEMERAL_CONTACT = 0.1  # people per day
 
 
 class CityPopulation(FixedNetworkPopulation):
+    def reset_people(self, society):
+        """
+        Override reset_people() to allow the new attribute person.home in CityPopulation can be saved when reset the
+        population according to different type of society
+        :param society:
+        :return:
+        """
+        # logging.info("CityPopulation.reset_people")
+        # count = 0
+        for person in self.people:
+            # if count < 5:
+            # logging.info(f"reset person's home to {person.home.type}, {person.home.coordinate['lon']} and {person.home.coordinate['lat']}")
+            # count += 1
+            person.__init__(society, config=society.cfg.__dict__, name=person.name, home=person.home)
+
     def fix_cliques(self, encounter_size):
         """
         :param encounter_size: not used
         :return:
         """
-        static_cliques = build_city_cliques(self.people)
+        static_cliques = self.build_city_cliques()
         logging.info(f"Adding {len(static_cliques)} permanent contact groups")
         dynamic_cliques = FixedNetworkPopulation.fix_cliques(self, EPHEMERAL_CONTACT)
         logging.info(f"Adding {len(dynamic_cliques)} ephemeral contact pairs")
         return static_cliques + dynamic_cliques
 
 
-def build_city_cliques(people):
-    """
-    :param people: a list of population.covid.PersonCovid() objects
-    :return: a list of little sets, each is a 'clique' in the graph, some are households, some are workplaces
-    each individual should belong to exactly one household and one workplace
-    for example: [{person_0, person_1, person_2}, {person_0, person_10, person_54, person_88, person_550, person_270}]
-    - except not everyone is accounted for of course
-    """
-    households = build_households(people)
-    report_size(households, 'households')
+    def build_city_cliques(self):
```

Comment:
I don't understand what's changed here. The side-by-side is not working very well ...

Reply:
Here I changed the function to a method of CityPoplation Class. This is to make sure that every home instance can be passed to each person of self.people, and save it there. I'm not sure if build_city_cliques(self.people) is passing a reference to self.people or just a copy. I will validate it later.